pkg/reconciler/vertex/scaling/scaling.go

```diff
@@ -130,6 +130,7 @@ func (s *Scaler) scale(ctx context.Context, id int, keyCh <-chan string) {
 // scaleOneVertex implements the detailed logic of scaling up/down a vertex.
 //
 // For source vertices which have both rate and pending message information,
+// if there are multiple partitions, we consider the max desired replicas among all partitions.
```

Comment:
Do we need to update the logic in back pressure check?

Reply:
https://github.com/numaproj/numaflow/blob/49a61569e637c2350152a693fd78a5a5a8b69fc7/pkg/reconciler/vertex/scaling/scaling.go#L255